File: python/2739.py
class Solution:
    def distanceTraveled(self, mainTank: int, additionalTank: int) -> int:
        ans = 0
        while additionalTank > 0 and mainTank >= 5:
            t = mainTank // 5
            ans += t * 5
            mainTank -= t * 5
            t = min(t, additionalTank)
            additionalTank -= t
            mainTank += t
        ans += mainTank
        return 10 * ans

File: go/2739.go
package main

func distanceTraveled(mainTank int, additionalTank int) int {
	res := 0
	for mainTank >= 5 && additionalTank > 0 {
		t := mainTank / 5
		mainTank %= 5
		res += 5 * t
		t = min(additionalTank, t)
		additionalTank -= t
		mainTank += t
	}
	res += mainTank
	return 10 * res
}
